src/extension.ts

```diff
@@ -784,41 +785,52 @@ function launchMetals(
     client.onRequest(MetalsSlowTask.type, (params, requestToken) => {
       return new Promise((requestResolve) => {
         const showLogs = ` ([show logs](command:${ClientCommands.ToggleLogs} "Show Metals logs"))`;
-        window.withProgress(
-          {
-            location: ProgressLocation.Notification,
-            title: params.message + showLogs,
-            cancellable: true,
-          },
-          (progress, progressToken) => {
-            // Update total running time every second.
-            let seconds = params.secondsElapsed || 0;
-            const interval = setInterval(() => {
-              seconds += 1;
-              progress.report({ message: readableSeconds(seconds) });
-            }, 1000);
-
-            // Hide logs and clean up resources on completion.
-            function onComplete() {
-              clearInterval(interval);
-            }
 
-            // Client triggered cancelation from the progress notification.
-            progressToken.onCancellationRequested(() => {
-              onComplete();
-              requestResolve({ cancel: true });
-            });
+        // Wait a bit before showing the progress notification
+        const waitTime = 2;
```

Comment:
Let's make sure that if there were already more than 2 seconds elapsed that we show the notification, otherwise we will be always adding it.

```suggestion
        const waitTime = 2;
        let delay = Math.max(0, waitTime - (params.secondsElapsed || 0))
```

Reply:
Good point. Done.